ocs_ci/deployment/metallb.py

```diff
@@ -0,0 +1,623 @@
+        metallb_inst_data = templating.load_yaml(constants.METALLB_INSTANCE_YAML)
+        metallb_inst_data.get("metadata").update({"namespace": self.namespace_lb})
+
+        metallb_inst_file = tempfile.NamedTemporaryFile(
+            mode="w+", prefix="metallb_instance", delete=False
+        )
+        templating.dump_data_to_temp_yaml(metallb_inst_data, metallb_inst_file.name)
+
+        exec_cmd(f"oc create -f {metallb_inst_file.name}", timeout=240)
+
+        return self.metallb_instance_created()
+
+    def create_ip_address_pool(self):
+        """
+        Create IP address pool for MetalLB
+
+        Returns:
+            bool: True if IP address pool is created, False if creation failed
+
+        Raises:
+            NotImplementedError: if platform is not supported
+            ValueError: if number of reserved IP addresses for MetalLB is not specified
+        """
+
+        # common part for both platforms
+        ipaddresspool_data = templating.load_yaml(METALLB_IPADDRESSPOOL_PATH)
+        if self.namespace_lb != METALLB_DEFAULT_NAMESPACE:
+            ipaddresspool_data.get("metadata").update({"namespace": self.namespace_lb})
+
+        self.ip_address_pool_name = ipaddresspool_data.get("metadata").get("name")
+
+        if self.ip_address_pool_created():
+            logger.info(
+                f"IPAddressPool {self.ip_address_pool_name} already exists in the namespace {self.namespace_lb}"
+            )
+            return True
+
+        if config.ENV_DATA["platform"] == constants.HCI_VSPHERE:
+            reserved_ips_num = config.ENV_DATA.get("ips_to_reserve")
+            if not reserved_ips_num:
+                raise ValueError(
+                    "Number of reserved IP addresses for MetalLB is not specified"
+                )
+            else:
+                for num in range(1, reserved_ips_num + 1):
+                    self.hostnames.append(
+                        f"clustername-{config.ENV_DATA['cluster_name']}-{num}"
+                    )
+
+            # due to circular import error, import is here
+            from ocs_ci.deployment.vmware import assign_ips
+
+            logger.info("Reserving IP addresses from IPAM and Creating IP address pool")
+            self.addresses_reserved = assign_ips(hosts=self.hostnames)
+
+            ip_addresses_with_mask = [ip + "/32" for ip in self.addresses_reserved]
+            ipaddresspool_data.get("spec").update({"addresses": ip_addresses_with_mask})
+
+        elif config.ENV_DATA["platform"] == constants.HCI_BAREMETAL:
+            if "ip_address_pool" not in config.ENV_DATA:
+                raise ValueError("IP address pool is not specified in the config file")
+
+            ipaddresspool_data.get("spec").update(
+                {"addresses": config.ENV_DATA["ip_address_pool"]}
+            )
+        else:
+            raise NotImplementedError(
+                f"Platform {config.ENV_DATA['platform']} is not supported yet"
+            )
```

Comment:
I think that if provided, we can use the values from `config.ENV_DATA["ip_address_pool"]` for any platform (not only the `HCI_BAREMETAL`).
Maybe even the values from this configuration could have precedence before the logic of reserving IPs from IPAM for the vSphere platform. In other words, if the `config.ENV_DATA["ip_address_pool"]` configuration will be provided, it will use the values from the configuration and if not, it will try to use the logic specific for the platform - e.g. reserve the IPs in IPAM server - and if no such logic is available, than raise the exception that the ip address pool is not provided and automatic assignment is not implemented for that platform. With this approach, it will be quite flexible for implementing new platforms and also for debugging purposes and similar cases. WDYT?